### PR TITLE
Add Dockerfile for CI build

### PR DESCRIPTION
**Pull request by mykola-kobets-epam**

### CODE REVIEW

File: docker/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,25 @@
+FROM ubuntu:24.04 AS aos-core-build-base
+
+ENV DEBIAN_FRONTEND=noninteractive
+ENV PATH="/root/.local/bin:${PATH}"
+
+# Install basic build dependencies
```
Comment (al1img) [inline docker/Dockerfile:6]: Add empty line after comment as the comment is related to group of operators/commands.

File: docker/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+RUN apt-get update \
+    && apt-get install -y lcov cppcheck libsofthsm2 libsystemd-dev python3-pip\
+       pipx cmake git autotools-dev autoconf libtool pkg-config python3-jinja2 \
+    && apt-get clean && rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/*
+
+RUN pipx install conan
+
+# Build conan dependencies
```
Comment (al1img) [inline docker/Dockerfile:14]: Ditto

File: docker/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,25 @@
+FROM ubuntu:24.04 AS aos-core-build-base
```
Comment (al1img) [inline docker/Dockerfile:1]: Probably better to put in aos_core_common_cpp repo. Let's discuss.

File: docker/Dockerfile (hunk 1 above)
Comment (mykola-kobets-epam) [reply docker/Dockerfile:6]: done

File: docker/Dockerfile (hunk 2 above)
Comment (mykola-kobets-epam) [reply docker/Dockerfile:14]: done

File: docker/Dockerfile (hunk 3 above)
Comment (mykola-kobets-epam) [reply docker/Dockerfile:1]: done

**Comment by mykola-kobets-epam** [comment]
moving to aos-core-common-cpp

**Pull request closed without merging**